model.py

```diff
@@ -331,16 +309,25 @@ def _run(self, inputs, labels=None, device='CPU', device_ids=None):
                 feed[v.name] = labels[idx]
 
         endpoints = self._endpoints[self.mode]
-        fetch_list = endpoints['output'] + endpoints['loss']
+        fetch_list = endpoints['output']
+        if 'loss' in endpoints:
+            fetch_list = endpoints['output'] + endpoints['loss']
         num_output = len(endpoints['output'])
-        out = self._executor.run(
-            compiled_prog, feed=feed,
-            fetch_list=fetch_list)
+        out = self._executor.run(compiled_prog,
+                                 feed=feed,
+                                 fetch_list=fetch_list)
         if self.mode == 'test':
             return out[:num_output]
         else:
             return out[:num_output], out[num_output:]
 
+    def _get_loss(self, outputs):
+        assert self.model._loss_function
+        label_vars = [k.forward() for k in to_list(self.model._labels)]
```

Comment:
`model._labels` may also be a dict and processed accordingly.

Reply:
Now, Loss requires lists of outputs and labels as input.  Not like forward in Model, each input has a key.   There is no key-name for labels in Loss. If use dict, the order is uncertain.